### PR TITLE
Added a source for RSS feeds.

### PR DESCRIPTION
**Pull request by jorisquist**
Added a source for RSS feeds.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -48,12 +48,17 @@ val coreDependencies = Seq(
...
   "org.json4s" % "json4s-scalap_2.11" % "3.6.0-M2",
-  "org.json4s" % "json4s-jackson_2.11" % "3.6.0-M2"
+  "org.json4s" % "json4s-jackson_2.11" % "3.6.0-M2",
 
   // "com.jsuereth" %% "scala-arm" % "2.0",
   // "org.scala-lang.modules" % "scala-java8-compat_2.11" % "0.8.0",
   // "org.scala-lang.modules" %% "scala-async" % "0.9.7",
   // "io.reactivex" %% "rxscala" % "0.26.5"
+
+  // https://mvnrepository.com/artifact/org.scalaj/scalaj-http
```
Comment (joskuijpers) [inline build.sbt:57]: Cruft

File: src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        if (lastItem.isDefined)
+          x.pubDate.isBefore(lastItem.get.pubDate) || lastItem.get.guid == x.guid
+        else
+          false
+      })
+        .foreach(ctx.collect)
+
+      lastItem = Some(sortedItems.last)
+
+
```
Comment (joskuijpers) [inline src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala]: Too many newlines 😄

File: src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    XML.loadString(http.getResponse(url).body)
+  }
+
+  def xmlToRSSItem(node: scala.xml.Node): RSSItem = {
+    val title = (node \ "title").text
+    val description = (node \ "description").text
+    val link = (node \ "link").text
+    val guid = (node \ "guid").text
+
+    //Tue, 08 May 2018 08:59:00 GMT
```
Comment (joskuijpers) [inline src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala:62]: Cruft

File: src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    val formatter = DateTimeFormatter.RFC_1123_DATE_TIME //.ofPattern("EEE, dd MMMMM yyyy HH:mm:ss z")
```
Comment (joskuijpers) [inline src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala:63]: Cruft

File: src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    val pubDate = LocalDateTime.parse((node \ "pubDate").text, formatter)
+
+    RSSItem(title, description, link, pubDate, guid)
+  }
+
```
Comment (joskuijpers) [inline src/main/scala/org/codefeedr/plugins/rss/RSSItemSource.scala:68]: Cruft

File: src/main/scala/org/codefeedr/utilities/HttpRequest.scala [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package org.codefeedr.utilities
+
+import scalaj.http.{Http, HttpResponse}
+
+class Http() {
```
Comment (joskuijpers) [inline src/main/scala/org/codefeedr/utilities/HttpRequest.scala:5]: Please explain why this class is needed.

File: src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import org.scalamock.function.FunctionAdapter1
+import org.scalatest.{BeforeAndAfter, FunSuite}
+import org.scalamock.scalatest.MockFactory
+import scalaj.http.HttpResponse
+
+import scala.io.Source
+
+class RSSItemSourceTest extends FunSuite with MockFactory with BeforeAndAfter{
+
+
```
Comment (joskuijpers) [inline src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala:15]: Cruft

File: src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    (ctxMock.collect _).expects(*).anyNumberOfTimes()
+
+    rssItemSource.open(null)
+    rssItemSource.run(ctxMock)
+  }
+
+  test("RSS source should collect all RSS items"){
+
```
Comment (joskuijpers) [inline src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala]: Cruft

File: src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    var rssItemList: List[RSSItem] = List()
+    (ctxMock.collect _).expects(new FunctionAdapter1[RSSItem, Boolean]((x:RSSItem) => {rssItemList = x :: rssItemList; true})).repeated(15)
+
+    rssItemSource.open(null)
+    rssItemSource.run(ctxMock)
+  }
+
+  test("RSS source should collect RSS items in order"){
+
```
Comment (joskuijpers) [inline src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala]: Cruft

File: src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+  test("RSS source should poll maxNumberOfRuns times") {
+    val httpMock = mock[Http]
+    val fakeUrl = "http://www.example.com"
+    val rssItemSource = new RSSItemSource(fakeUrl, 2000, 2, httpMock)
+
+    val ctxMock = mock[SourceFunction.SourceContext[RSSItem]]
+
+    val rssResponsesName = "/RSSItemSourceTestResponses"
```
Comment (joskuijpers) [inline src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala:23]: Maybe add some comments?

File: src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    val lines = Source.fromURL(getClass.getResource(rssResponsesName)).getLines
+
+    for (line <- lines) {
+      val response = HttpResponse[String](line, 0, null)
+      (httpMock.getResponse _).expects(*).returning(response).noMoreThanOnce()
+    }
+
+    (ctxMock.collect _).expects(*).anyNumberOfTimes()
```
Comment (joskuijpers) [inline src/test/scala/org/codefeedr/rss/RSSItemSourceTest.scala]: Expects what?

**Review by joskuijpers — CHANGES_REQUESTED** [review]

**Comment by jorisquist** [comment]
Could you review the new changes to see if the branch can be merged now?

**Review by joskuijpers — APPROVED** [review]

**Pull request merged**